client/signup/main.jsx

```diff
@@ -345,6 +346,7 @@ class Signup extends Component {
 			intent: get( signupDependencies, 'intent' ),
 			starting_point: get( signupDependencies, 'startingPoint' ),
 			is_in_hosting_flow: hostingFlow,
+			...( mainFlowName ? { flow: mainFlowName } : {} ),
```

Comment:
Maybe I'm missing something, but could we use the `queryObject?.flow` here instead of passing a new prop to the component?

Reply:
Thanks, I've updated it here https://github.com/Automattic/wp-calypso/pull/84191/commits/0b9c51ca07ca23e1f360ca1a4c3485822fb291b8